ci-tools/host-runner/README.md

```diff
@@ -0,0 +1,255 @@
+```
+
+### non-NixOS
+
+1. Install Nix
+1. Install cross-compilation tools
+    - E.g. Debian, `sudo apt-get update -q -y && sudo apt-get install -q -y qemu-system-aarch64 binfmt-support qemu-user-static`
+1. Update your Nix config to the following
+
+```bash
+cat << EOF > /etc/nix/nix.conf
+sandbox = true
+
+experimental-features = nix-command flakes
+trusted-users = $USER
+extra-platforms = aarch64-linux
+extra-sandbox-paths = /usr/bin/qemu-aarch64-static
+EOF
+```
+
+# Bootstrapping a new Raspberry PI
+
+## Config modifications
+
+It's likely that you will want to make the following modifications to the NixOS configuration.
+
+### Add SSH public key as an Authorized User
+
+See `configuration.nix`.
+
+```nix
+    # Add your ssh public key as a string to this list.
+    # Remove keys you don't trust.
+    openssh.authorizedKeys.keys = [
+        # clundin Mac
+        "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDDKTJ6unwymfvdFSTNAXo+wjaX1l2SFPgeSgK/xzC7ex3oGR2ihCg/8luQt1e6FKnbqV83O2v0AT/aRw9p9sEjY7HGNDz+0nQ6lezi4XAuqJMOMshzVlqv4hZJLb8Ab2PMma0se15h1LhnfSUpttv7cDgdLXHqh2kizMQ39l62Lu4j2ITJKFhqW1v7Ez74uo2o++We6EHU2PRZhyKV9tKbYXojOyow+abUXKMfXy01iCSunaQq6KRB6Jl5TskMVmGSz0rUnjyxLCCPEA2h7D0lgQviLuJQtIl/jFYu8QFNqaVwHDHiEUpNfcfQGx6S7hpSs7CdPD29YQSka9TovICyD3dCKGn+tpfRQDmZSTR8Qnqv4mNtxKPcitpMFNVL9V6Echqy83rlo5CgO1tEsL/6g0WEm6nrFBMs/szUfv1qs4/4wL0PsNit1ArxfqYXVaDzGisvA+Y4yRl2IsMPaI7TzB6uDSR0j31jZXSGR8vqPG9rF+aGobF21OfWGHI8Ddc= clundin@clundin-macbookpro.roam.internal"
+        # clundin Workstation
+        "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDohnyJgm7nztWkxtKaqds13IHJMYpoV2VozRs5wIbct2R98lmyATIR+pOypPv/uv++KnDTUV68/Pt+SFZS6VcOBj/SfDqniqi5/Zmj5qL0dRfLfr4RE1ET7gMPMpvbynUEaXiaochSInikdToDwUeUfhNfs1JGGIbOsoNNhBYAKuNTBo7DXpOUuq8t6oBMqvYWWtCN+kAagkf3tyi94Br52GS++9i7q3RZnvHtw79FW5Sc4xZtuiBqs7aKsK/pplKC7V6emcf0zM1F49knZR+UmLvHhRXzbyxJPyDHgZFGcu7SeDCikAn/2mKxIr8gHjyVSZ5JwOuHekQrVRgwT+CVYBc6AsvCe5aRsrUDJC7TZHsWUVKkaXXDyASYy87wTy+IE3BCZVZmjjZ9OufX7jqXT/lenJbaOn9240e5pSydOzT97tVIuL8rRL+6m00cBsxLJcsFjmrGreX3M2T37IifICpFUDaZJYkVcrKvUWXIdgqHECpQgo7YmLlTcC/hP/0= clundin@clundin6.kir.corp.google.com "
+      ];
+```
+
+##  Flash SD Image
+Create a new NixOS image. You should modify the flake target to the host your are bootstrapping. This example uses "caliptra-hostrunner0".
+
+```bash
+$ nix run nixpkgs#nixos-generators -- -f sd-aarch64 --flake path:.#caliptra-hostrunner0 --system aarch64-linux -o ./hostrunner-image
+...
+$ ls hostrunner-image/sd-image # A new RPI image is created in the hostrunner-image folder. De-compress with zst and then flash the SD card for the Raspberry PI.
+nixos-sd-image-24.11.20250421.9684b53-aarch64-linux.img.zst
+```
+
+Once the image is flashed the Raspberry PI should be ready to boot, assuming secrets were available when generating the SD image.
+
+## Remotely modify hostrunner
+
+You can cross-compile a NixOS system and copy binaries using SSH. Below is an example command to cross-compile the caliptra-hostrunner0.
```

Comment:
Is this an alternative to flashing the SD image? It currently reads like both steps are required.